osu.Game.Modes.Taiko/UI/InputDrum.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Input;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Framework.Graphics.Transforms;
+using osu.Framework.Input;
+using osu.Game.Graphics;
+using System.Collections.Generic;
+
+namespace osu.Game.Modes.Taiko.UI
+{
+    /// <summary>
+    /// A component of the playfield that captures input and displays input as a drum.
+    /// </summary>
+    internal class InputDrum : Container
+    {
+        public InputDrum()
+        {
+            Size = new Vector2(TaikoPlayfield.PlayfieldHeight);
+
+            Children = new Drawable[]
+            {
+                new TaikoHalfDrum(false)
+                {
+                    Name = "Left Half",
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.CentreRight,
+                    RelativeSizeAxes = Axes.Both,
+                    Keys = new List<Key>(new[] { Key.F, Key.D })
+                },
+                new TaikoHalfDrum(true)
+                {
+                    Name = "Right Half",
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.CentreLeft,
+                    RelativeSizeAxes = Axes.Both,
+                    Position = new Vector2(-1f, 0),
+                    Keys = new List<Key>(new[] { Key.J, Key.K })
+                }
+            };
+        }
+
+        /// <summary>
+        /// A half-drum. Contains one centre and one rim hit.
+        /// </summary>
+        private class TaikoHalfDrum : Container
+        {
+            /// <summary>
+            /// A list of keys which this half-drum accepts.
+            /// <para>
+            /// [0] => Inner key, [1] => Outer key
+            /// </para>
+            /// </summary>
+            public List<Key> Keys = new List<Key>();

```

Comment:
not sure this should be a list if it only ever accepts two keys. just have `InnerKey` and `OuterKey`?